src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -400,10 +400,23 @@ protected void paintIcon(final Graphics g, final AbstractButton b, final Rectang
 
         if (icon == null) return;
 
+        Icon selectedIcon = null;
```

Comment:
It looks like a similar pattern is used in the AquaButtonLabeledUI.paintIcon(), can we align the logic of both methods?